plugins/inputs/linux_cpu/linux_cpu.go

```diff
@@ -0,0 +1,188 @@
+// +build linux
+
+package linux_cpu
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const defaultHostSys = "/sys"
+
+type LinuxCPU struct {
+	Log             telegraf.Logger `toml:"-"`
+	PathSysfs       string          `toml:"host_sys"`
+	GatherCPUFreq   bool            `toml:"gather_cpufreq"`
+	GatherThrottles bool            `toml:"gather_throttles"`
+	cpus            []cpu
+}
+
+type cpu struct {
+	id    string
+	path  string
+	props map[string]*os.File
+}
+
+type prop struct {
+	name     string
+	path     string
+	optional bool
+}
+
+var sampleConfig = `
+  ## Path for sysfs filesystem.
+  ## See https://www.kernel.org/doc/Documentation/filesystems/sysfs.txt
+  ## Defaults:
+  # host_sys = "/sys"
+
+  ## Gather CPU frequency information
+  ## Defaults:
+  # gather_cpufreq = true
+
+  ## Gather CPU throttle information per core
+  ## Defaults:
+  # gather_throttles = false
+`
+
+func (g *LinuxCPU) SampleConfig() string {
+	return sampleConfig
+}
+
+func (g *LinuxCPU) Description() string {
+	return "Collects CPU metrics exposed on Linux"
+}
+
+func (g *LinuxCPU) Init() error {
+	if g.PathSysfs == "" {
+		g.PathSysfs = defaultHostSys
+	}
+
+	cpus, err := g.discoverCpus()
+	if err != nil {
+		return err
+	}
+	g.cpus = cpus
```

Comment:
Theoretically, we may have `len(g.cpus) == 0`.

Shouldn't we fail fast in that case right here?